.github/workflows/ci-repo.yml

```
@@ -0,0 +1,81 @@
+name: CI (repo level)
+
+on:
+  push:
+    branches:
+      - "main"
+      - "release-*"
+
+  pull_request:
+    branches:
+      - "main"
+      - "release-*"
+
+  workflow_dispatch:
+
+  workflow_call:
+
+concurrency:
+  group: ci-repo-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  package-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check-out repository
+        uses: actions/checkout@v4
+
+      - name: Use Node.js from .nvmrc
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: "vscode/.nvmrc"
+
+      - name: Cache extension node_modules
+        uses: actions/cache@v3
+        with:
+          path: vscode/node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('vscode/package-lock.json') }}
+
+      - name: Install dependencies
+        run: |
+          cd vscode
+          npm version
+          npm ci
```

I imagine it might look a little less silly, although not practically different if you did this.

```suggestion
      - name: Install dependencies
        working-directory: ./vscode
        run: |
          npm version
          npm ci
```

---

That's also an option. Simple `cd vscode` was the fastest fix.  But there is more then one place where we have this kind of problems. We should probably decide how we want to proceed with our kind-of-monorepo.